modules/atmosphere/rendering/atmospheredeferredcaster.cpp

```diff
@@ -0,0 +1,1595 @@
+        << "Mie Extinction coeff: " << glm::to_string(_mieExtinctionCoeff) << std::endl
+        << "Rayleigh Scattering coeff: " << glm::to_string(_rayleighScatteringCoeff) << std::endl
+        << "Mie Scattering coeff: " << glm::to_string(_mieScatteringCoeff) << std::endl;
+    std::cout << ss.str() << std::endl;
+
+    //==========================================================
+    //========= Load Shader Programs for Calculations ==========
+    //==========================================================
+    loadComputationPrograms();
+    
+    //==========================================================
+    //============ Create Textures for Calculations ============
+    //==========================================================
+    createComputationTextures();
+
+    // Saves current FBO first
+    GLint defaultFBO;
+    glGetIntegerv(GL_FRAMEBUFFER_BINDING, &defaultFBO);
+
+    GLint m_viewport[4];
+    glGetIntegerv(GL_VIEWPORT, m_viewport);
+
+    // Creates the FBO for the calculations
+    GLuint calcFBO;
+    glGenFramebuffers(1, &calcFBO);
+    glBindFramebuffer(GL_FRAMEBUFFER, calcFBO);
+    GLenum drawBuffers[1] = { GL_COLOR_ATTACHMENT0 };
+    glDrawBuffers(1, drawBuffers);
+
+    // Prepare for rendering/calculations
+    GLuint quadCalcVAO;
+    GLuint quadCalcVBO;
+    createRenderQuad(&quadCalcVAO, &quadCalcVBO, 1.0f);
+
+    // Starting Calculations...
+    LDEBUG("Starting precalculations for scattering effects...");
+
+    //==========================================================
+    //=================== Execute Calculations =================
+    //==========================================================
+    executeCalculations(quadCalcVAO, drawBuffers, 6);
+
+    deleteUnusedComputationTextures();
+
+    // Restores system state
+    glBindFramebuffer(GL_FRAMEBUFFER, defaultFBO);
+    glViewport(m_viewport[0], m_viewport[1],
+        m_viewport[2], m_viewport[3]);
+    glDeleteBuffers(1, &quadCalcVBO);
+    glDeleteVertexArrays(1, &quadCalcVAO);
+    glDeleteFramebuffers(1, &calcFBO);
+
+    LDEBUG("Ended precalculations for Atmosphere effects...");
+}
+
+
+
+void AtmosphereDeferredcaster::resetAtmosphereTextures()
+{
+
+}
+
+void AtmosphereDeferredcaster::createRenderQuad(GLuint * vao, GLuint * vbo,
+                                                const GLfloat size) {
+
+    glGenVertexArrays(1, vao);
+    glGenBuffers(1, vbo);
+    glBindVertexArray(*vao);
+    glBindBuffer(GL_ARRAY_BUFFER, *vbo);
+
+    const GLfloat vertex_data[] = {
+        //      x      y     z     w
+        -size, -size, 0.0f, 1.0f,
+        size,    size, 0.0f, 1.0f,
+        -size,  size, 0.0f, 1.0f,
+        -size, -size, 0.0f, 1.0f,
+        size, -size, 0.0f, 1.0f,
+        size,    size, 0.0f, 1.0f
+    };
+
+    glBufferData(GL_ARRAY_BUFFER, sizeof(vertex_data), vertex_data, GL_STATIC_DRAW);
+    glVertexAttribPointer(0, 4, GL_FLOAT, GL_FALSE, sizeof(GLfloat) * 4, reinterpret_cast<GLvoid*>(0));
```

Comment:
I am following the declaration here: [glVertexAttribPointer](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glVertexAttribPointer.xhtml)